index.d.ts

```diff
@@ -0,0 +1,209 @@
+ * any of the events defined in the `rejectionEvents` option.
+ *
+ * **Note**: `event` is a string for a single event type, for example, `'data'`. To listen on multiple
+ * events, pass an array of strings, such as `['started', 'stopped']`.
+ *
+ * The returned promise has a `.cancel()` method, which when called, removes the event listeners and causes the promise to never be settled.
+ *
+ * @param emitter Event emitter object. Should have either a `.on()`/`.addListener()`/`.addEventListener()` and
+ * `.off()`/`.removeListener()`/`.removeEventListener()` method, like the [Node.js `EventEmitter`](https://nodejs.org/api/events.html) and
+ * [DOM events](https://developer.mozilla.org/en-US/docs/Web/Events).
+ * @param event Name of the event or events to listen to. If the same event is defined both here and in
+ * `rejectionEvents`, this one takes priority.
+ */
+declare function pEvent<EmittedType, EmittedTypeRest = EmittedType>(
+	emitter: Emitter<EmittedType, EmittedTypeRest>,
+	event: string | symbol | (string | symbol)[],
+	options: MultiArgumentsOptions<EmittedType | EmittedTypeRest>
+): CancelablePromise<(EmittedType | EmittedTypeRest)[]>;
+declare function pEvent<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
+	event: string | symbol | (string | symbol)[],
+	filter: FilterFunction<EmittedType>
+): CancelablePromise<EmittedType>;
+declare function pEvent<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
+	event: string | symbol | (string | symbol)[],
+	options?: Options<EmittedType>
+): CancelablePromise<EmittedType>;
+
+export default pEvent;
+
+/**
+ * Wait for multiple event emissions. Returns an array.
+ */
+export function multiple<EmittedType, EmittedTypeRest = EmittedType>(
+	emitter: Emitter<EmittedType, EmittedTypeRest>,
+	event: string | symbol | (string | symbol)[],
+	options: MultipleMultiArgumentsOptions<EmittedType | EmittedTypeRest>
+): CancelablePromise<(EmittedType | EmittedTypeRest)[][]>;
+export function multiple<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
+	event: string | symbol | (string | symbol)[],
+	options: MultipleOptions<EmittedType>
+): CancelablePromise<EmittedType[]>;
+
+/**
+ * Returns an [async iterator](http://2ality.com/2016/10/asynchronous-iteration.html) that lets you asynchronously
+ * iterate over events of `event` emitted from `emitter`. The iterator ends when `emitter` emits an event matching
+ * any of the events defined in `resolutionEvents`, or rejects if `emitter` emits any of the events defined in
+ * the `rejectionEvents` option.
+ */
+export function iterator<EmittedType, EmittedTypeRest = EmittedType>(
+	emitter: Emitter<EmittedType, EmittedTypeRest>,
+	event: string | symbol | (string | symbol)[],
+	options: IteratorMultiArgumentsOptions<EmittedType | EmittedTypeRest>
+): AsyncIterableIterator<(EmittedType | EmittedTypeRest)[]>;
+export function iterator<EmittedType>(
+	emitter: Emitter<EmittedType, any>,
```

Comment:
Use `unknown`.

Reply:
This is the type we accept, not the type we expose to the user. It should be `any` to signal that we accept anything.

Reply:
`unknown` accepts anything too. Anything can be assigned to it.

Reply:
Sorry, I didn't really explain what I meant.

My problem with `unknown` is two-fold:

1. It is a relatively new feature, not everyone knows about it.
2. If we use this type, it raises the minimum compatible TS version to 3.0. If we don't use any other >= TS 3.0 features, it is actually regrettable to lose support for older versions of TS just because of this. The places you've annotated don't actually benefit from usage of `unknown`, `any` has the same semantic meaning there.

Do you still want me to change that?

Reply:
1. It's almost a year ago since it was released. Also, devs are expected to keep up with the language. I'm not gonna *not* use new features just in case some people don't know those features.
2. I don't see that as a concern. `unknown` was released a long time ago.

> Do you still want me to change that?

Yes